CHANGELOG.md

```diff
@@ -42,6 +42,42 @@ Breaking changes:
 
 - [core] fixed typo (highligh -> highlight) in caption highlight fragment [#7050](https://github.com/eclipse-theia/theia/pull/7050)
 - [terminal] added new abstract methods to the TerminalWidget[#7179]: `scrollLineUp`, `scrollLineDown`, `scrollToTop`, `scrollPageUp`, `scrollPageDown`
+- The release includes the removal of language-specific Theia extensions and other Theia extensions that are or can be replaced by equivalent VS Code extensions.
+  - Migration steps are available at the following wiki page [`Consuming Builtin and External VS Code Extensions`](https://github.com/eclipse-theia/theia/wiki/Consuming-Builtin-and-External-VS-Code-Extensions).
+  - Alternatively, it is also possible to use [`yarn resolutions`](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) to keep using old native extensions.\
+    For example, in your application's `package.json`, you can include the following (to keep using `@theia/typescript` and `@theia/textmate-grammars`):
+    ```json
+    "resolutions": {
```

Comment:
I think this might need a bit more work. e.g. If we take the browser example and try to pin a couple of compatible extensions to `0.14.0`, we'll end-up with a non-working app because we'll have a mix of `0.14.0` and `0.15.0`. 

To be sure this works, I think it's necessary to pin all theia extensions to `0.14.0`. 

Can you give this a quick try and update if needed?

Reply:
something like here with globs: https://github.com/theia-ide/theia-apps/blob/b70f8cf84830b7724acdf5f5315edef6b5b675ed/theia-ruby-docker/latest.package.json#L30 should help

Reply:
i.e. `"**/@theia/typescript": "0.15.0"`

Reply:
I'm wondering, should I take out the guesswork here and just point to the `yarn resolutions` page like I've done? I can update the wiki as well to show examples with resolutions.

Reply:
> I'm wondering, should I take out the guesswork here and just point to the `yarn resolutions` page like I've done?

The tricky part is avoiding having multiple versions of any Theia extension being included in the app - the generic documentation will not help much with that. 

> I can update the wiki as well to show examples with resolutions.

Maybe we can test using a glob to pin all Theia extensions at once and if it works have that as example? Or alternatively just mention that it may be necessary to pin all Theia extensions and add a concrete example to the wiki ASAP.

Reply:
@marcdumais-work I updated the resolution, it ended up working for me to do so.

```json
"resolutions": {
  "**/@theia/**": "0.14.0",
},
```

Reply:
It basically says use 0.14.0 for everything 😆 So maybe we just tell that since 0.15.0 these extensions are deprecated and VS Code counterparts should be used.

Reply:
> It basically says use 0.14.0 for everything 😆 So maybe we just tell that since 0.15.0 these extensions are deprecated and VS Code counterparts should be used.

I'd be fine with that, what about you? @marcdumais-work

Reply:
0.15.0 would be better as example post-release, but would not have been a good test right now.

> It basically says use 0.14.0 for everything

True, but at least this could be used temporarily by extenders to keep building their app against 0.14.0 or 0.15.0 after we release 0.16.0, which I think would not work without the resolutions block. It can buy them time until they can adapt to use the VS Code extensions?

Reply:
> I'd be fine with that, what about you? @marcdumais-work

I think GH is having issues ATM - it took over an hour for my previous comment to be posted. Since we're running out of time I suggest we go with Anton's suggestion, but mentioning `0.15.0` instead of `0.14.0`, and merge ASAP. If necessary we can elaborate about keep using an older Theia version, in the wiki.

Reply:
I mean with `"**/@theia/**": "0.14.0"` there is no point to upgrade to 0.15.0, since it tells don't use 0.15.0 use 0.14.0, there is no upgrade. They could just keep using 0.14.0 as a version without any yarn resolutions.

Reply:
test

Reply:
Anton: I just tested and you're right: if I take the browser example and ask for `0.14.0` for every theia extension, the resulting app ends-up working without a resolutions block. Similar had not worked for me in the past, but if it work now, no need to document a useless work-around.